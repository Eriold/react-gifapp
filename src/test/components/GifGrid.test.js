import React from 'react'
import { shallow } from 'enzyme'
import { GifGrid } from '../../Components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchifs';

jest.mock('../../hooks/useFetchifs')


describe('render <GifGrid />', () => {
  const category = 'Death Note';

  test('show component ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot();
  })

  test('show items when load images useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/any/any.jpg',
      title: 'test gif'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })


})