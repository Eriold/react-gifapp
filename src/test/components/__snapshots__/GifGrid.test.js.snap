// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GifGrid /> show component  1`] = `
<Fragment>
  <h3>
    Death Note
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`render <GifGrid /> show items when load images useFetchGifs 1`] = `
<Fragment>
  <h3>
    Death Note
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test gif"
      url="https://localhost/any/any.jpg"
    />
  </div>
</Fragment>
`;
